source/ConversationPanel.cpp

```diff
@@ -437,10 +448,6 @@ void ConversationPanel::Goto(int index, int selectedChoice)
 // Exit this panel and do whatever needs to happen next.
 void ConversationPanel::Exit()
 {
-	// Finish the PlayerInfo transaction so any changes get saved again.
-	if(useTransactions)
-		player.FinishTransaction();
-
```

Comment:
Could moving the FinishTransaction() to the destructor of the ConversationPanel be problematic if the BoardingPanel on line 468 decides to start a transaction?

And the callback on line 473 is apparently capable of forcing takeoff, causing the savegame on takeoff to be done without the transaction finished?

And is the callback on line 473 able to start a new conversation (and thus a new transaction)?

I guess it is also possible for multiple ConversationPanels to pop up after each other in a single gamestep? In that case we should also not delay the finishing of the transaction to the next gamestep.